### PR TITLE
feat(Auth): Added BrandwatchReactAuth component

### PR DESCRIPTION
**Pull request by HHogg**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
-  "globals": {
-    "document": true,
-    "module": true,
-    "__dirname": true,
-    "require": true,
-  },
   "env": {
-    "jest": true,
+    "amd": true,
+    'browser': true,
```
Comment (thatguynamedandy) [inline .eslintrc]: oh really minor, but single quotes here. (good call on using env property though).

File: src/components/App.js [hunk 2]
```diff
@@ -17,12 +20,48 @@ export default class App extends Component {
...
   };
 
+  static contextTypes = {
+    brandwatchAuthGetProfile: PropTypes.func.isRequired,
+    brandwatchAuthLogout: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (thatguynamedandy) [inline src/components/App.js]: In a real life scenario I guess we would probably push this to redux?

File: src/components/App.test.js [hunk 3]
```diff
@@ -4,10 +4,17 @@ import sinon from 'sinon';
...
-  return shallow(<App { ...props } />);
+  return shallow(<App { ...props } />, {
+    context: {
+      brandwatchAuthGetProfile: authGetProfileStub,
+      brandwatchAuthLogout: authLogoutStub,
+    },
+  });
 }
 
 beforeEach(() => {
```
Comment (thatguynamedandy) [inline src/components/App.test.js]: When testing with context I almost prefer to treat it exactly the same as props. I guess these stubs wouldn't be reset before each test which could lead to issues in the future. What do you think of the below?

```js
function render(props, options = {}) {
  return shallow(<App { ...props } />, options);
}

let props;
let options;

beforeEach(() => {
  props = {
    ...
  }
  options = {
    context: {
      ...
    }
  }
});

test('blah', () => {
  render(props, options);
});

File: src/components/App.js (hunk 2 above)
Comment (pmsorhaindo) [reply src/components/App.js]: Is there a scenario where we wouldn't want the auth profile in the store?

Comment (HHogg) [reply src/components/App.js]: I've but it in local state for now as @thatguynamedandy saga stuff incoming changes redux a bit. He's agreed to then handle this.

**Pull request merged**